test/python/test_user_config.py

```diff
@@ -20,128 +20,52 @@
 
 
 class TestUserConfig(QiskitTestCase):
-    @classmethod
-    def setUpClass(cls):
-        cls.file_path = "temp.txt"
+    def setUp(self):
+        super().setUp()
+        current_dir = os.path.dirname(os.path.abspath(__file__))
+        test_name = str(self.id()).split(".")[-1]
+        file_path = os.path.join(current_dir, "configurations", test_name + ".conf")
+        self.config = user_config.UserConfig(file_path)
```

Comment:
I'm not super thrilled about this approach. Personally I find this kind of fragile to have to match the test name with a filename in a relative directory. Also as you mentioned it moves the configuration away from the test code which makes it harder to develop and debug tests here. I do wonder if there is an approach we can use that avoids shared state but also lets us create files.

What about something like:
```python
def setUp(self):
    super().setUp()
    self.file_path = 'test_%s.conf' % uuid.uuid4()
```

and then leave tests as they were. That would ensure a unique file_name for each test. The only downside with that approach is there would need to be a cleanup in each test to delete the file for each test (but that was already done before).

Reply:
Oh doh. Thanks, I am unsure why the grouping change wasn't included in that commit. I must have done something funky when I rebased on main.

Yea I considered using a uuid to create a filename. My gut was sort of against it, but I do not  really have any specific reasons for it. I think I just have a hesitation of writing files in a test unless it is necessary.   If you are fine with the uuid then that is enough for me to get over it though.